### ISSUE TITLE
Undoing a destroy Roster leaves the roster with no admin

### ISSUE THREAD

**Issue by bandogora** [umts/screaming-dinosaur#169]
When you click the "undo" link after destroying a Roster the Roster is recreated from it's last version, but it's associations are not. 
Memberships are destroyed when their Roster is:
https://github.com/umts/screaming-dinosaur/blob/da4a878d10f7dfb518acb8303e39d33c96ddfd43/app/models/roster.rb#L6

This means we loose the Membership that defines what User is an admin for that Roster. With no admin, no User is allowed to edit or destroy that Roster.

I see two possible ways of fixing this:
- [Saving association data](https://github.com/paper-trail-gem/paper_trail#4b-associations) with the Roster version and versioning Memberships.
- Adding an admin column to `Roster` , this would require changing the association between Users and Rosters and probably removing the Membership class.

**Comment by werebus** [comment]
* Don't bother allowing "undo" of roster deletion?

I dunno, this has the feeling of something that's going to be a big pain to unravel. And, practically speaking, no one has ever destroyed a roster in "real life".

**Comment by jforgue** [comment]
> * Don't bother allowing "undo" of roster deletion?

Yeah, that works too. I don't think it would be _too_ hard to unravel. But I agree, adding a confirm to the "Destroy" button is probably good enough for rosters.
